doc/nrf/libraries/networking/downloader.rst

```diff
@@ -0,0 +1,195 @@
+.. _lib_downloader:
+
+Downloader
+##########
+
+.. contents::
+   :local:
+   :depth: 2
+
+You can use the downloader client library to download files from a server.
+
+Overview
+********
+
+The download is carried out in a separate thread and the application receives events such as :c:enumerator:`DOWNLOADER_EVT_FRAGMENT` that contain the data fragments as the download progresses.
+When the download completes, the library sends the :c:enumerator:`DOWNLOADER_EVT_DONE` event to the application.
+
+Protocols
+=========
+
+The library supports HTTP, HTTPS (TLS 1.2), CoAP, and CoAPS (DTLS 1.2) over IPv4 and IPv6.
+If other protocols are required, they can be added by the application.
+See :file:`downloader_transport.h` for details.
+The protocol used for the download is specified in the beginning of the ``URI`` or ``host``.
+Use ``http://`` for HTTP, ``https://`` for HTTPS, ``coap://`` for CoAP, and ``coaps://`` for CoAPS.
+If no protocol is specified, the downloader defaults to HTTP or HTTPS, depending on the server security configuration.
+
+.. _downloader_https:
+
+HTTP and HTTPS (TLS 1.2)
+------------------------
+
+When downloading using HTTP, the library sends only one HTTP request to the server and receives only one HTTP response.
+
+When downloading using HTTPS with an nRF91 Series device, it is carried out through `Content-Range requests (IETF RFC 7233)`_ due to memory constraints that limit the maximum HTTPS message size to two kilobytes.
+The library thus sends and receives as many requests and responses as the number of fragments that constitute the download.
+For example, to download a file of 47 kilobytes with a fragment size of two kilobytes, a total of 24 HTTP GET requests are sent.
```

Comment:
For consistency

```suggestion
For example, to download a file of 47 kilobytes with a fragment size of 2 kilobytes, a total of 24 HTTP GET requests are sent.
```

Reply:
I've already changed this from `2` to `two` based on tech writers comments.

Reply:
@divipillai I think that in this case `2` instead of `two` is preferred since there is a math calculation implied in this sentence.

Reply:
updated